src/core/ipv4/ip4.c

```diff
@@ -448,11 +574,35 @@ ip4_forward(struct pbuf *p, struct ip_hdr *iphdr, struct netif *inp)
 static int
 ip4_input_accept(struct netif *netif)
 {
-  LWIP_DEBUGF(IP_DEBUG, ("ip_input: iphdr->dest 0x%"X32_F" netif->ip_addr 0x%"X32_F" (0x%"X32_F", 0x%"X32_F", 0x%"X32_F")\n",
-                         ip4_addr_get_u32(ip4_current_dest_addr()), ip4_addr_get_u32(netif_ip4_addr(netif)),
-                         ip4_addr_get_u32(ip4_current_dest_addr()) & ip4_addr_get_u32(netif_ip4_netmask(netif)),
-                         ip4_addr_get_u32(netif_ip4_addr(netif)) & ip4_addr_get_u32(netif_ip4_netmask(netif)),
-                         ip4_addr_get_u32(ip4_current_dest_addr()) & ~ip4_addr_get_u32(netif_ip4_netmask(netif))));
+
+#if IP_DEBUG
+  u32_t a, b, c, d, e, f;
+  a = ip4_addr_get_u32(ip4_current_dest_addr());
+  b = ip4_addr_get_u32(netif_ip4_addr(netif));
+  c = ip4_addr_get_u32(ip4_current_dest_addr()) & ip4_addr_get_u32(netif_ip4_netmask(netif));
+  d = ip4_addr_get_u32(netif_ip4_addr(netif)) & ip4_addr_get_u32(netif_ip4_netmask(netif));
+  e = ip4_addr_get_u32(ip4_current_dest_addr()) & ~ip4_addr_get_u32(netif_ip4_netmask(netif));
+  f = ip4_addr_get_u32(ip4_current_src_addr());
+
+  LWIP_DEBUGF(IP_DEBUG, ("ip4_input: iphdr->src %"U16_F".%"U16_F".%"U16_F".%"U16_F" ",
+			   ((const u8_t*) (&f))[0], ((const u8_t*) (&f))[1],
+			   ((const u8_t*) (&f))[2], ((const u8_t*) (&f))[3]));
```

Comment:
Could this be simplified, e.g. using?
```suggestion
  ip4_addr_debug_print_val(IP_DEBUG, ip4_current_src_addr());
```